demos/defender/defender_demo_json/metrics_collector.c

```diff
@@ -378,4 +378,152 @@ MetricsCollectorStatus_t GetEstablishedConnections( Connection_t * pOutConnectio
 
     return status;
 }
+
+/*-----------------------------------------------------------*/
+
+MetricsCollectorStatus_t GetCpuUsageStats( CpuUsageStats_t * pCpuUsage )
+{
+    MetricsCollectorStatus_t status = MetricsCollectorSuccess;
+    FILE * fileHandle = NULL;
+    uint32_t filledVariables;
+    char lineBuffer[ MAX_LINE_LENGTH ];
+
+    if( pCpuUsage == NULL )
+    {
+        LogError( ( "Invalid parameter. pCpuUsage %p", ( void * ) pCpuUsage ) );
+        status = MetricsCollectorBadParameter;
+    }
+
+    if( status == MetricsCollectorSuccess )
+    {
+        fileHandle = fopen( "/proc/uptime", "r" );
+
+        if( fileHandle == NULL )
+        {
+            LogError( ( "Failed to open /proc/uptime." ) );
+            status = MetricsCollectorFileOpenFailed;
+        }
+    }
+
+    if( status == MetricsCollectorSuccess )
+    {
+        if( fgets( &( lineBuffer[ 0 ] ), MAX_LINE_LENGTH, fileHandle ) != NULL )
+        {
+            float upTime = 0.0f, idleTime = 0.0f;
+            LogDebug( ( "File: /proc/uptime, Content: %s.",
+                        &( lineBuffer[ 0 ] ) ) );
+
+            /* Parse the output. */
+            filledVariables = sscanf( &( lineBuffer[ 0 ] ),
+                                      "%f %f",
```

Comment:
Do we really need to use float here? The code that outputs these numbers utilizes %lu: https://elixir.bootlin.com/linux/v4.0/source/fs/proc/uptime.c#L27

```c
            filledVariables = sscanf( &( lineBuffer[ 0 ] ),
                                      "%lu.%*u %lu.%*u",
                                      &( upTime ),
                                      &( idleTime ) );
```

Reply:
Updated to use integer type then for the non-decimal part of the value. 
Based on the source code link, we may have to ideally use `struct timespec` (to store higher precision of data) type but for simplicity of reporting only the seconds part of the time, we can use integer type.